tests/switchless_one_tcs/host/host.c

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <inttypes.h>
+#include <limits.h>
+#include <openenclave/host.h>
+#include <openenclave/internal/atomic.h>
+#include <openenclave/internal/error.h>
+#include <openenclave/internal/tests.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "../../../host/hostthread.h"
+#include "../../../host/strings.h"
+#include "switchless_one_tcs_u.h"
+
+#if defined(__linux__)
+#include <unistd.h>
+#elif defined(_WIN32)
+#include <windows.h>
+#endif
+
+static status_e switchless_func_status = FUNC_NOT_START;
```

Comment:
Changing this to `static volatile status_e switchless_func_status` would prevent the hang in release mode.

Reply:
@yanxue-intel Can you incorporate the volatile change requested by @anakrish? we can then merge this PR

Reply:
ok

Reply:
@anakrish  Thanks.

Reply:
> Changing this to `static volatile status_e switchless_func_status` would prevent the hang in release mode.

I can’t define a volatile variable inside the EDL file.
`static volatile status_e switchless_func_status` Ecall also needs this variable.
`        public void enc_empty_switchless(
            [user_check] status_e *status ) transition_using_threads;
`

Reply:
In that case, you could change 
`while (switchless_func_status != FUNC_WORKING)`
to 
`while (*(volatile status_e*)&switchless_func_status != FUNC_WORKING)`

Reply:
ok,thanks